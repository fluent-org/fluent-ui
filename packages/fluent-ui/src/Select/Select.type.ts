import * as React from 'react'

export interface SelectProps
  extends StandardProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement, 'onChange'> {
  value?: string | number
  onChange?: (value: string | number) => void
  disabled?: boolean
  children: React.ReactElement | React.ReactElement[]
}
