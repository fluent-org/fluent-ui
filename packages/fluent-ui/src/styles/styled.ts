import styled from 'styled-components'
import { th, variant, system, breakpoints } from '@xstyled/system'

export { styled, system, th, variant, breakpoints }

export * from 'styled-components'
export * from '@xstyled/system'
// export * from '@xstyled/prop-types'

export * from './stylize'
