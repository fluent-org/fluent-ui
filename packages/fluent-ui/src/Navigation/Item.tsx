import * as React from 'react'
import { styled, css, variant, th } from '../styles/styled'
import Box from '../Box'
import { NavigationContext } from './Navigation'
import * as CSS from 'csstype'

export type ID = string | number

interface ItemProps extends Omit<React.ComponentPropsWithoutRef<'div'>, 'id'>, StyledProps {
  id?: ID
  title?: string
  onClick?: React.MouseEventHandler<HTMLDivElement>
  children: React.ReactNode
}

type Child = React.ReactChild | any

const StyledItemWrapper = styled(Box)<{
  reveal: boolean
}>`
  position: relative;
  cursor: pointer;
  text-decoration: none;
  max-height: 40px;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 12px;
  box-sizing: border-box;
  transition: ${th.transition('navigation')};
  color: ${th.color('black.default')};
  background-color: ${({ reveal }): CSS.ColorProperty =>
    reveal ? th.color('standard.light2') : 'transparent'};
  &:hover {
    background-color: ${({ reveal }): CSS.ColorProperty =>
      reveal ? th.color('standard.light1') : th.color('standard.transparent1')};
  }
  &:active {
    color: ${th.color('black.default')};
    background-color: ${({ reveal }): CSS.ColorProperty =>
      reveal ? th.color('standard.dark1') : th.color('standard.transparent2')};
  }
`
const StyledItemActiveBar = styled.div<{
  active: boolean
  horizontal: boolean
}>`
  position: absolute;
  background-color: ${th.color('primary.default')};
  transition: ${th.transition('navigation')};
  ${variant({
    prop: 'horizontal',
    variants: {
      true: css`
        width: 80%;
        height: 4px;
        left: 10%;
        bottom: 0;
      `,
      false: css`
        width: 6px;
        height: 24px;
        left: 0;
        top: 50%;
        margin-top: -12px;
      `
    }
  })}
  ${variant({
    prop: 'active',
    variants: {
      true: css`
        transform: scale3d(1, 1, 1);
      `,
      false: css`
        transform: scale3d(0, 0, 0);
      `
    }
  })}
`
const StyledItemIconWrapper = styled.div`
  height: 100%;
  transition: ${th.transition('navigation')};
`
const StyledItemTextWrapper = styled.div<{ expanded: boolean }>`
  width: ${({ expanded }): string => (expanded ? 'auto' : '0px')};
  opacity: ${({ expanded }): number => (expanded ? 1 : 0)};
  padding-left: ${({ children, expanded }): string =>
    children ? (expanded ? '12px' : '0px') : '0px'};
  flex: 1;
  align-items: center;
  overflow: hidden;
  white-space: nowrap;
  transition: ${th.transition('navigation')};
`

const Item: React.FC<ItemProps> = ({
  id,
  onClick,
  children,
  ...rest
}: ItemProps): React.ReactElement => {
  const container: {
    icon: any
    content: any
  } = {
    icon: null,
    content: null
  }
  React.Children.forEach(
    children,
    (child: Child): void => {
      if (
        child.type &&
        child.type.displayName &&
        String.prototype.includes.call(child.type.displayName, 'FIcon')
      ) {
        container.icon = React.cloneElement(child, {
          style: { width: 16, height: 16 }
        })
      } else {
        container.content = React.cloneElement(child, {
          style: { fontSize: 14 }
        })
      }
    }
  )

  // handle active item
  const { value: activeID, onChange, expanded, reveal, horizontal } = React.useContext(
    NavigationContext
  )
  function handleItemClick(e: React.MouseEvent<HTMLDivElement>): void {
    onClick && onClick(e)
    onChange && onChange(id as ID)
  }
  const [active, setActive] = React.useState(false)
  React.useEffect((): void => {
    if (id) {
      if (activeID === id) setActive(true)
      else if (activeID && activeID !== id) setActive(false)
    }
  }, [activeID, id])

  return (
    <StyledItemWrapper onClick={handleItemClick} reveal={reveal} {...rest}>
      {!!id && <StyledItemActiveBar active={active} horizontal={horizontal} />}
      <StyledItemIconWrapper>{container.icon}</StyledItemIconWrapper>
      <StyledItemTextWrapper expanded={expanded}>{container.content}</StyledItemTextWrapper>
    </StyledItemWrapper>
  )
}

Item.displayName = 'FNavigationItem'

export default Item
