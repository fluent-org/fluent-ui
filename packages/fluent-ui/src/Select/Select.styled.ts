import { Style, Styles } from 'jss'
import { Theme } from '../styles'
import { SelectClassProps } from './Select.type'

const root: Style = {
  display: 'inline-block',
  position: 'relative',
  minWidth: 160,
  height: 32,
  font: 'inherit',
  cursor: 'pointer'
}
const disabled: Style = {
  cursor: 'not-allowed'
}

const select = (theme: Theme): Style => ({
  outline: 'none',
  font: 'inherit',
  width: '100%',
  height: '100%',
  cursor: 'pointer',
  borderRadius: 2,
  border: '2px solid',
  paddingLeft: 12,
  paddingRight: 32,
  display: 'flex',
  alignItems: 'center',
  borderColor: theme.colors!.standard!.default,
  transition: theme.transitions!.input,
  '&:hover': {
    borderColor: theme.colors!.standard!.dark1
  },
  '&:active, &:focus': {
    borderColor: theme.colors!.primary!.default
  }
})
const selectDisabled = (theme: Theme): Style => ({
  color: theme.colors!.standard!.dark2,
  backgroundColor: theme.colors!.standard!.light1,
  cursor: 'not-allowed',
  '&:hover, &:active, &:focus': {
    borderColor: theme.colors!.standard!.default
  }
})

const icon = (theme: Theme): Style => ({
  width: 32,
  height: '100%',
  position: 'absolute',
  right: 0,
  top: 0,
  fontSize: 12,
  display: 'inline-flex',
  justifyContent: 'center',
  alignItems: 'center',
  // disabledIcon false
  color: theme.colors!.standard!.dark3,
  cursor: 'pointer'
})
const disabledIcon = (theme: Theme): Style => ({
  color: theme.colors!.standard!.dark2,
  cursor: 'not-allowed'
})

export const styles = (theme: Theme): Styles<SelectClassProps> => ({
  root,
  disabled,
  select: select(theme),
  selectDisabled: selectDisabled(theme),
  icon: icon(theme),
  disabledIcon: disabledIcon(theme)
})
