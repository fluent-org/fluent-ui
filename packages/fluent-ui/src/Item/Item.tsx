import * as React from 'react'
import { NavigationContext } from '../Navigation/Navigation'
import { ListContext } from '../List/List'
import {
  StyledItemWrapper,
  StyledItemActiveBar,
  StyledItemPrefixWrapper,
  StyledItemTextWrapper
} from './Item.styled'
import { NavigationID } from '../Navigation/Navigation.type'
import { ItemProps } from './Item.type'

const Item = React.forwardRef<HTMLDivElement, ItemProps>(
  ({ value, prefix, active, children, onClick, ...rest }: ItemProps, ref): React.ReactElement => {
    // handle active item
    const {
      value: activeID,
      onChange,
      expanded,
      reveal: navigationReveal,
      horizontal
    } = React.useContext(NavigationContext)
    const { reveal: listReveal } = React.useContext(ListContext)
    const reveal = navigationReveal || listReveal

    const handleItemClick = React.useCallback(
      (e: React.MouseEvent<HTMLDivElement>): void => {
        onClick && onClick(e)
        onChange && value && onChange(value as NavigationID)
      },
      [value, onChange, onClick]
    )
    const [_active, setActive] = React.useState(false)
    React.useEffect((): void => {
      if (value && activeID) {
        if (activeID === value) {
          setActive(true)
        } else {
          setActive(false)
        }
      }
    }, [activeID, value])
    return (
      <StyledItemWrapper ref={ref} onClick={handleItemClick} reveal={reveal} {...rest}>
        {!!value && <StyledItemActiveBar active={active || _active} horizontal={horizontal} />}
        <StyledItemPrefixWrapper>{prefix}</StyledItemPrefixWrapper>
        <StyledItemTextWrapper expanded={expanded} hasPrefix={!!prefix}>
          {children}
        </StyledItemTextWrapper>
      </StyledItemWrapper>
    )
  }
)

Item.displayName = 'FItem'

export default Item
