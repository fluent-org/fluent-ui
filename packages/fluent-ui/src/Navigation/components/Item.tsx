import * as React from 'react'
import { NavigationContext } from '../Navigation'
import {
  StyledItemWrapper,
  StyledItemActiveBar,
  StyledItemIconWrapper,
  StyledItemTextWrapper
} from './Item.styled'
import { NavigationItemProps, NavigationID } from '../Navigation.type'

type Child = React.ReactChild | any

const Item = ({ id, onClick, children, ...rest }: NavigationItemProps): React.ReactElement => {
  const container: {
    icon: any
    content: any
  } = {
    icon: null,
    content: null
  }
  React.Children.forEach(
    children,
    (child: Child): void => {
      if (
        child.type &&
        child.type.displayName &&
        String.prototype.includes.call(child.type.displayName, 'FIcon')
      ) {
        container.icon = React.cloneElement(child, {
          style: { width: 16, height: 16 }
        })
      } else {
        container.content = React.cloneElement(child, {
          style: { fontSize: 14 }
        })
      }
    }
  )

  // handle active item
  const { value: activeID, onChange, expanded, reveal, horizontal } = React.useContext(
    NavigationContext
  )
  function handleItemClick(e: React.MouseEvent<HTMLDivElement>): void {
    onClick && onClick(e)
    onChange && onChange(id as NavigationID)
  }
  const [active, setActive] = React.useState(false)
  React.useEffect((): void => {
    if (id) {
      if (activeID === id) setActive(true)
      else if (activeID && activeID !== id) setActive(false)
    }
  }, [activeID, id])

  return (
    <StyledItemWrapper onClick={handleItemClick} reveal={reveal} {...rest}>
      {!!id && <StyledItemActiveBar active={active} horizontal={horizontal} />}
      <StyledItemIconWrapper>{container.icon}</StyledItemIconWrapper>
      <StyledItemTextWrapper expanded={expanded}>{container.content}</StyledItemTextWrapper>
    </StyledItemWrapper>
  )
}

Item.displayName = 'FNavigationItem'

export default Item
