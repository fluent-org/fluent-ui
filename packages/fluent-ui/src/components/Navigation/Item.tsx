import React, {
  ReactElement,
  ReactNode,
  Children,
  ReactComponentElement,
  ReactChild,
  cloneElement,
  useContext,
  useEffect,
  useState
} from 'react'
import Box from '../Box'
import { Icon } from '@fluent-ui/icons'
import styled from '@xstyled/styled-components'
import { th } from '@xstyled/system'
import { NavigationContext } from './Navigation'
import { useDispatch } from '../../hooks/useAction'

export type ID = string | number

interface ItemProps {
  id?: ID
  title?: string
  children: ReactNode
}

type Child = ReactComponentElement<typeof Icon> | ReactChild | any

const StyledItemWrapper = styled.div`
  position: relative;
  max-height: 40px;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 12;
  box-sizing: border-box;
  transition: ${th.transition('navigation')};
  &:hover {
    background-color: secondary;
  }
  &:active {
    background-color: primary;
  }
`
interface StyledItemActiveBarProps {
  active: boolean
}
const StyledItemActiveBar = styled.div<StyledItemActiveBarProps>`
  position: absolute;
  left: 0;
  top: 50%;
  background-color: accent;
  width: 6px;
  height: 24px;
  margin-top: -12px;
  transform: ${({ active }): string =>
    active ? 'scale3d(1,1,1)' : 'scale3d(0,0,0)'};
  transition: ${th.transition('navigation')};
`

const Item = ({ id, children }: ItemProps): ReactElement => {
  const container: {
    icon: any
    content: any
  } = {
    icon: null,
    content: null
  }
  Children.forEach(
    children,
    (child: Child): void => {
      if (child.type && child.type.displayName === 'FIcon') {
        container.icon = cloneElement(child, {
          style: { width: 16, height: 16 }
        })
      } else {
        container.content = cloneElement(child, {
          style: { fontSize: 14 }
        })
      }
    }
  )

  // handle active item
  const activeID = useContext(NavigationContext)
  const dispatch = useDispatch({ type: 'navigation/handleActive', payload: id })
  function handleItemClick(): void {
    dispatch()
  }
  const [active, setActive] = useState(false)
  useEffect((): void => {
    if (id) {
      if (activeID === id) setActive(true)
      else if (activeID && activeID !== id) setActive(false)
    }
  }, [activeID, id])

  return (
    <StyledItemWrapper onClick={handleItemClick}>
      {id && <StyledItemActiveBar active={active} />}
      <Box marginRight={12} height="100%">
        {container.icon}
      </Box>
      <Box flex={1} display="flex" alignItems="center" overflow="hidden">
        {container.content}
      </Box>
    </StyledItemWrapper>
  )
}

export default Item
