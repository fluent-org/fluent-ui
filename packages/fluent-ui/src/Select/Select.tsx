import * as React from 'react'
import classNames from 'classnames'
import { createUseStyles } from '@fluent-ui/styles'
import { styles } from './Select.styled'
import { Theme } from '../styles'

import { usePopper, useClick, useClickOutside } from '@fluent-ui/hooks'
import { ChevronDownMed as DownIcon } from '@fluent-ui/icons'
import { SelectClassProps, SelectProps, SelectPropTypes } from './Select.type'
import Transition from '../Transition'
import List from '../List'

export const name = 'Select'

const useStyles = createUseStyles<Theme, SelectClassProps>(styles, { name })

const Select: React.FC<SelectProps> = React.forwardRef<HTMLInputElement, SelectProps>(
  (props, ref): React.ReactElement => {
    const {
      as: Component = 'span',
      className: classNameProp,
      value,
      onChange,
      disabled = false,
      children,
      ...rest
    } = props

    const [referenceRef, popperRef] = usePopper<HTMLDivElement, HTMLDivElement>({
      placement: 'bottom-start',
      positionFixed: false
    })

    const [visible, setVisible] = React.useState(false)
    const handleVisible = React.useCallback(
      (status): void => {
        disabled !== true && setVisible(status)
      },
      [disabled]
    )
    const [_, bind] = useClick(handleVisible)
    useClickOutside(
      referenceRef,
      (): void => {
        setVisible(false)
      }
    )

    const [currentText, setCurrentText] = React.useState('')
    React.useEffect((): void => {
      React.Children.forEach(
        children,
        (child: React.ReactElement): void => {
          const active = child.props.value === value
          active && setCurrentText(child.props.children)
        }
      )
    }, [children, value])
    const theChildren = React.Children.map(
      children,
      (child: React.ReactElement): React.ReactElement => {
        function onClick(): void {
          child.props.value && onChange && onChange(child.props.value)
          setCurrentText(child.props.children)
          setVisible(false)
        }
        const active = child.props.value === value
        return React.cloneElement(child, {
          onClick,
          active
        })
      }
    )

    const classes = useStyles(props)
    const className = classNames(
      classes.root,
      {
        [classes.disabled]: disabled
      },
      classNameProp
    )
    const selectClassName = classNames(classes.select, {
      [classes.selectDisabled]: disabled
    })
    const iconClassName = classNames(classes.icon, {
      [classes.disabledIcon]: disabled
    })

    return (
      <Component className={className} ref={referenceRef} {...bind}>
        <button className={selectClassName}>{currentText}</button>
        <span className={iconClassName}>
          <DownIcon />
        </span>
        <input type="hidden" value={value} disabled={disabled} ref={ref} {...rest} />
        <Transition type="grow" visible={visible} wrapper={false} mountOnEnter unmountOnExit>
          <List zIndex={1002} ref={popperRef}>
            {theChildren}
          </List>
        </Transition>
      </Component>
    )
  }
)

Select.displayName = `F${name}`

Select.propTypes = SelectPropTypes

Select.defaultProps = {
  disabled: false
}

export default Select
