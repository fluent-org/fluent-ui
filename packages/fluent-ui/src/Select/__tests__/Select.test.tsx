import * as React from 'react'
import { getClasses, render } from '../../test-utils'
import '@testing-library/jest-dom/extend-expect'

import Select, { name } from '../Select'
import { styles } from '../Select.styled'
import { SelectClassProps } from '../Select.type'

import Item from '../../Item'

const classes = getClasses<SelectClassProps>(styles, name)

describe('Select', (): void => {
  const testText = 'Hello World'

  test('should be support ref', (): void => {
    const ref = React.createRef<HTMLInputElement>()
    const { container } = render(
      <Select ref={ref}>
        <Item>{testText}</Item>
      </Select>
    )
    const { current: element } = ref
    const RadioElement = container.querySelector(`input`) as HTMLInputElement
    expect(element).toEqual(RadioElement)
  })

  test('should render children', (): void => {
    const { container, getByText, sheets } = render(
      <Select>
        <Item>{testText}</Item>
      </Select>
    )
    expect(getByText(testText)).toBeInTheDocument()
    expect(container.firstChild).toHaveClass(classes.root)
    expect(sheets.toString()).toMatchSnapshot()
  })

  test('should be render a disabled Select', (): void => {
    const ref = React.createRef<HTMLInputElement>()
    const { container } = render(
      <Select ref={ref} disabled={true}>
        <Item>{testText}</Item>
      </Select>
    )
    expect(container.firstChild).toHaveClass(classes.disabled)
    expect(ref.current!.disabled).toEqual(true)
  })
})
