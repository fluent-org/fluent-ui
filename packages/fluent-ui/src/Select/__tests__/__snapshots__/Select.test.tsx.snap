// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render children 1`] = `
".FTransition-root {}
.FTransition-fade {}
.FTransition-fade.fade-enter {
  opacity: 0;
}
.FTransition-fade.fade-exit {
  opacity: 1;
}
.FTransition-fade.fade-exit-active {
  opacity: 0;
}
.FTransition-fade.fade-enter-active {
  opacity: 1;
}
.FTransition-zoom {}
.FTransition-zoom.zoom-enter {}
.FTransition-zoom.zoom-exit {}
.FTransition-zoom.zoom-exit-active {
  transform: scale3d(0, 0, 0);
}
.FTransition-zoom.zoom-exit-done {
  transform: scale3d(0, 0, 0);
}
.FTransition-zoom.zoom-enter-active {
  transform: scale3d(0, 0, 0);
}
.FTransition-zoom.zoom-enter-done {
  transform: none;
}
.FTransition-slide {}
.FTransition-slide.slide-enter {
  transform: translateY(100vh);
}
.FTransition-slide.slide-exit {
  transform: none;
}
.FTransition-slide.slide-exit-active {
  transform: translateY(100vh);
}
.FTransition-slide.slide-exit-done {
  transform: translateY(100vh);
}
.FTransition-slide.slide-enter-done {
  transform: none;
}
.FTransition-collapse {}
.FTransition-grow {}
.FTransition-grow.grow-enter {
  opacity: 0;
  transform: scale(0.75, 0.5625);
}
.FTransition-grow.grow-exit {}
.FTransition-grow.grow-exit-active {
  opacity: 0;
  transform: scale(0.75, 0.5625);
}
.FTransition-grow.grow-exit-done {
  opacity: 0;
  transform: scale(0.75, 0.5625);
}
.FTransition-grow.grow-enter-active {
  opacity: 1;
  transform: scale(1, 1);
}
.FTransition-grow.grow-enter-done {
  opacity: 1;
  transform: none;
}
.FTransition-root-0 {
  opacity: 0;
  transform: scale(0.75, 0.5625);
  transition: opacity 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,transform 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,visibility 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms;
  visibility: hidden;
  transform-origin: 0 0 0;
}
.FTransition-collapse-1 {}
.FTransition-collapse-1.collapse-enter {
  height: 0;
}
.FTransition-collapse-1.collapse-exit {
  height: 0;
}
.FTransition-collapse-1.collapse-exit-active {
  height: 0;
}
.FTransition-collapse-1.collapse-enter-active {
  height: 0;
}
.FTransition-collapse-1.collapse-enter-done {
  height: auto;
}
.FTransition-collapse-1.collapse-enter {
  height: 0;
}
.FTransition-collapse-1.collapse-exit {
  height: 0;
}
.FTransition-collapse-1.collapse-exit-active {
  height: 0;
}
.FTransition-collapse-1.collapse-enter-active {
  height: 0;
}
.FTransition-collapse-1.collapse-enter-done {
  height: auto;
}
.FSelect-root {
  font: inherit;
  cursor: pointer;
  height: 32px;
  display: inline-block;
  position: relative;
  min-width: 160px;
}
.FSelect-disabled {
  cursor: not-allowed;
}
.FSelect-select {
  font: inherit;
  width: 100%;
  border: 2px solid;
  cursor: pointer;
  height: 100%;
  display: flex;
  outline: none;
  transition: background-color 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,color 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms,border-color 250ms cubic-bezier(0.76, 0.24, 0.23, 0.94) 0ms;
  align-items: center;
  border-color: #dedede;
  padding-left: 12px;
  border-radius: 2px;
  padding-right: 32px;
}
.FSelect-select:hover {
  border-color: #b5b5b5;
}
.FSelect-select:active, .FSelect-select:focus {
  border-color: #0078d4;
}
.FSelect-selectDisabled {
  color: #8c8c8c;
  cursor: not-allowed;
  background-color: #e9e9e9;
}
.FSelect-selectDisabled:hover, .FSelect-selectDisabled:active, .FSelect-selectDisabled:focus {
  border-color: #dedede;
}
.FSelect-icon {
  top: 0;
  color: #595959;
  right: 0;
  width: 32px;
  cursor: pointer;
  height: 100%;
  display: inline-flex;
  position: absolute;
  font-size: 12px;
  align-items: center;
  justify-content: center;
}
.FSelect-disabledIcon {
  color: #8c8c8c;
  cursor: not-allowed;
}"
`;
